app/src/main/java/org/wikipedia/analytics/metricsplatform/MetricsPlatform.kt

```diff
@@ -0,0 +1,133 @@
+package org.wikipedia.analytics.metricsplatform
+
+import org.wikimedia.metrics_platform.MetricsClient
+import org.wikimedia.metrics_platform.context.AgentData
+import org.wikimedia.metrics_platform.context.ClientData
+import org.wikimedia.metrics_platform.context.MediawikiData
+import org.wikimedia.metrics_platform.context.PerformerData
+import org.wikipedia.WikipediaApp
+import org.wikipedia.auth.AccountUtil
+import org.wikipedia.settings.Prefs
+import org.wikipedia.util.ReleaseUtil
+import java.util.Random
+
+object MetricsPlatform {
+    private val agentData: AgentData = AgentData(
+        WikipediaApp.instance.appInstallID,
+        "mobile app",
+        "android"
+    )
+
+    private val mediawikiData: MediawikiData = MediawikiData(
+        null,
+        WikipediaApp.instance.versionCode.toString(),
+        ReleaseUtil.isProdRelease,
+        ReleaseUtil.isDevRelease,
+        WikipediaApp.instance.wikiSite.dbName(),
+        WikipediaApp.instance.languageState.systemLanguageCode,
+        WikipediaApp.instance.languageState.appLanguageLocalizedNames
+    )
+
+    private val performerData: PerformerData = PerformerData(
```

Comment:
The `performerData` and `mediawikiData` objects contain items that can easily change at runtime, such as `userName`, `isLoggedIn`, `wikiSite`, `appLanguageCode`, etc.

Ideally these should not be constructed statically for the lifetime of the process, and instead should be retrieved via callbacks on demand from within the library.

Reply:
per our convo, nulling out these property groups (performer, mediawiki) from the MetricsClient constructor and passing in ApplicationData as part of every event submission - hopefully latest commit addresses correctly